### ISSUE TITLE
Stopped working lg c2 

### ISSUE THREAD

**Issue by Matteofar10** [JPersson77/LGTVCompanion#199]
Been working fine for awhile just stopped so re scanned and added again fresh test works to turn off but not to come back on when enter is pressed shut down doesn't turn off the c2 either and doesn't come on when pc is booted will post log and screen shots of settings now too

**Comment by Matteofar10** [comment]
![Screenshot 2024-03-01 160655](https://github.com/JPersson77/LGTVCompanion/assets/161858625/ab8e5423-8712-403e-9e34-ff4d2cdb208e)
![Screenshot 2024-03-01 160705](https://github.com/JPersson77/LGTVCompanion/assets/161858625/6dc9b625-19c0-4a3e-adb8-fca13f06a4af)
![Screenshot 2024-03-01 160723](https://github.com/JPersson77/LGTVCompanion/assets/161858625/1667a50d-4187-4bc2-bb61-464de0e07522)
![Screenshot 2024-03-01 160805](https://github.com/JPersson77/LGTVCompanion/assets/161858625/efddfde1-f1e2-4b22-8caa-de97d4283d9a)

**Comment by JPersson77** [comment]
Hi, from your logs I can see that you have tried various things - I notice you have two devices with the same IP so maybe delete one of them.

Can you try something for me please. Confirm that the IP and mac-address of your TV's network adapter (wired and wireless have different) correspond to the value in the device configuration. Then revert to option no 2 (device IP-adsress) and see how that works. If this does not work then please go to c:/programdata/lgtv companion/ and copy log.txt and config.json and paste a link here. You can use  pastebin.com for example

**Comment by Matteofar10** [comment]
i have 1 lg device it always finds 2 on the scan but 1 has no details in and other has the subnet and ip address in already i do have a 7" screen within my pc tower (not LG) i will try that not and post it asap thanks for the rapid response to

**Comment by Matteofar10** [comment]
just noticed inactive screen time any key press brings the C2 back on but not sure if thats pc settings or the lg companion app too

**Comment by Matteofar10** [comment]
[Log.txt](https://github.com/JPersson77/LGTVCompanion/files/14465202/Log.txt)

**Comment by JPersson77** [comment]
Probably the scan has picked up wrong data. 

It looks like the wake-on-lan is failing by looking at the log. So you need to confirm that the mac-address is correct by comparing the app configuration with the network adapter on the TV, whether it is wireless or wired.
![image](https://github.com/JPersson77/LGTVCompanion/assets/77928210/5a16704b-48c4-4036-ae68-b6460cd97b58)

then you need to ensure the device can be woken using Wake-on-lan.

_Ensure that the TV can be woken via the network. For the CX line of displays this is accomplished by navigating to Settings (cog button on remote)->All Settings->Connection->Mobile Connection Management->TV On with Mobile, and then enable 'Turn On via Wi-Fi'. For C1, C2 and C3 it's All Settings->General->Devices->External Devices->TV On With Mobile->Turn on via Wi-Fi. NOTE! This step is needed regardless of using WiFi or a cable._

**Comment by Matteofar10** [comment]
mac from tv is correct also have it set to wake via wifi and mobile ip from tv is correct pc is wired to router and tv using wifi for internet access used to work fine with subnet option (broadcast address according to device ip address and subnet mask) seems messing around with options gave different results but none gave fully working quality.. could it be the C2 settings maybe a clash somewhere always ready is on without wallpaper option and quick start + is enabled too... just i notice now the standby light used to always be red when shut down pc but now is white ?

**Comment by Matteofar10** [comment]
[Log.txt](https://github.com/JPersson77/LGTVCompanion/files/14468681/Log.txt)
![20240302_085636](https://github.com/JPersson77/LGTVCompanion/assets/161858625/97436311-67e7-4cee-9f13-b356d3d59591)
![20240302_085601](https://github.com/JPersson77/LGTVCompanion/assets/161858625/4204b15c-2925-4189-85b4-b9255ae1fa8e)
![20240302_085542](https://github.com/JPersson77/LGTVCompanion/assets/161858625/dce3027b-b47c-4e83-a0c5-5af25d7f349b)
![20240302_085155](https://github.com/JPersson77/LGTVCompanion/assets/161858625/8e7030c9-5763-4eb5-bc88-9b06164353ae)
![20240302_085153](https://github.com/JPersson77/LGTVCompanion/assets/161858625/94607e11-54fb-450a-ad91-ae1f889e0c56)
![20240302_085149](https://github.com/JPersson77/LGTVCompanion/assets/161858625/768fb08b-0928-4370-941e-495a6117f08c)
![20240302_085138](https://github.com/JPersson77/LGTVCompanion/assets/161858625/1448f043-6e75-477f-92cb-8b9ae69fad87)

**Comment by Matteofar10** [comment]
fresh log is attached with the pictures above too

**Comment by Matteofar10** [comment]
https://github.com/JPersson77/LGTVCompanion/assets/161858625/eb795413-5681-4adb-82da-606721f6d982


https://github.com/JPersson77/LGTVCompanion/assets/161858625/c4e6290e-0127-4b7f-a447-fa8dfffe0063

**Comment by Matteofar10** [comment]
two videos there one is ip setting (slow reaction and power up on its own no key press) other is subnet mask (fast reaction no power back on without remote)

**Comment by bootyacc** [comment]
Kinda having the same issue, using LG C3. startup and shutdown doesnt work. inactivity works. recently updated to 03.30.73

**Comment by JPersson77** [comment]
There's a wealth of information here, bit hard to follow.

You seem to be using wifi to connect the TV to the LAN. I think from your screenshots this means that the MAC-address is 20:28:BC:28:E8:BA and IP is IP 192.168.1.250. You seem to have configured this correctly in the last part of the log.

It seems the PC can communicate with the TV as turning it off with the test-function works for example.

But, it seems there are network issues of some kind for you. I can see in the last part of the attached log that it takes seconds for the PC to get a response from the TV. For example this:

`Sat 09:09:12 > -poweroff Device1
Sat 09:09:12 > [IPC] Force power OFF: Device1 
Sat 09:09:12 > Device1, spawning Thread_DisplayOff().
Sat 09:09:13 > -poweron Device1
Sat 09:09:13 > [IPC] Force power ON: Device1 
Sat 09:09:13 > Device1, omitted Thread_DisplayOn().
Sat 09:09:22 > Device1, [DEBUG] (SSL) OFF response 1: {"type":"registered","id":"register_0","payload":{"client-key":"5bf6a6e63caf5ab972260bae4f8f2b6b"}}
Sat 09:09:23 > Device1, [DEBUG] (SSL) OFF response 2: {"type":"response","id":2,"payload":{"state":"Active","returnValue":true}}
Sat 09:09:24 > Device1, [DEBUG] (SSL) OFF response 4: {"type":"response","id":3,"payload":{"returnValue":true}}
Sat 09:09:24 > Device1, power state is: OFF.`

The command to power off was issued at 09:09:12 but the response is registered a full ten seconds later when it should be immediate. Well, unless the TV was powered off at  the moment and you were trying to communicate with a device that was already powred off. Side note: keep "always ready" enabled on the TV so that it is possible to communicate with the TV even when it is off. In earlier parts of the log the response was immediate.

In other parts of the log it seems like the TV does not properly respond to the Wake-on-lan package. I know you have said that you have enabled this option in the TV but it seems unreliable. Again, can it be that your network is discarding WOL-packets? You could try to troybleshoot by using another tool and see if you can wake the TV, f.e. https://wol.aquilatech.com/

Have you tried using a cable instead of wifi? I've seen instances where problems like yours was resolved by restarting all network equipment, including access points, switches etc.

**Comment by JPersson77** [comment]
> Kinda having the same issue, using LG C3. startup and shutdown doesnt work. inactivity works. recently updated to 03.30.73

Hi, it will probably be easier to start a new issue. Please ensure to attach the log and config.json to the post

**Comment by JPersson77** [comment]
Hi any feedback regarding the above? Also try v4.0.0 to see if that helps

**Comment by JPersson77** [comment]
Closing due to missing feedback, please feel free to reopen as needed.
